### DISCUSSION TITLE
StaticFiles with follow_symlink not working correctly in venvs on Windows.

### DISCUSSION THREAD

**Discussion by falkoschindler** [encode/starlette#2111] [Potential Issue] [answered]
It looks like there is something wrong in `StaticFiles.loopup_path`. On venvs on Windows we noticed a problem with the common prefix check.

Consider the following test script (here I'm refering to our library called NiceGUI, but any other side package should give a similar result):

```py
from nicegui import ui

directory = Path(ui.__file__).parent / 'static'
print(f'{directory = }')
print(f'{os.path.abspath(directory)  = }')
print(f'{os.path.realpath(directory) = }')
```

This yields the following output:

```
directory = WindowsPath('C:/Users/username/Envs/venv_name/lib/site-packages/nicegui/static')
os.path.abspath(directory)  = 'C:\\Users\\username\\Envs\\venv_name\\lib\\site-packages\\nicegui\\static'
os.path.realpath(directory) = 'C:\\Users\\username\\Envs\\venv_name\\Lib\\site-packages\\nicegui\\static'
```

Note that `realpath` returns an uppercase "Lib" while `abspath` doesn't.

This causes the common prefix to be only "C:\Users\username\Envs\venv_name\", which is outside of the `directory`. Therefore the following check is true and the directory is not served:

https://github.com/encode/starlette/blob/f7bf74194358cceb5708f758989df89240c0c1aa/starlette/staticfiles.py#L172

I'm not perfectly sure how to fix it. Maybe using `commonpath` instead of `commonprefix`? Maybe converting strings to lowercase first? Maybe using pathlib instead of os.path?

Anyway, it's an issue which confused us quite a bit over here: https://github.com/zauberzeug/nicegui/issues/714

**Comment by Kludex** [comment, marked as answer]
I've created https://github.com/encode/starlette/issues/2112 from this discussion.
